practice/c/simple-char-driver.c

```
@@ -0,0 +1,23 @@
+// simple-char-driver.c -> for Linux. This is a simple character driver that initializes and exits cleanly, printing messages to the kernel log.
+// To compile this driver, you would typically create a Makefile and use the `make` command to build it against the Linux kernel headers.
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+
+static int __init simple_char_driver_init(void) {
```

Clang (14.0.6)

[warning] declaration uses identifier '__init', which is a reserved identifier

(bugprone-reserved-identifier)